controllers/utils/multicloud_util.go

```diff
@@ -302,25 +318,74 @@ func DeleteMultiCloudObjects(
 	}
 
 	for _, mcv := range mcvList.Items {
-		multiCloudLog.Info("[DeleteMultiClusterObjects] Delete ManagedClusterView", "name", mcv.Name)
+		multiCloudLog.Info("[DeleteMultiClusterObjects] Delete ManagedClusterView", "ns", mcv.Namespace, "name", mcv.Name)
 		if err := c.Delete(ctx, &mcv); err != nil {
 			return err
 		}
 	}
-
 	return nil
 }
 
-// DeleteAllMultiCloudObjects cleans up views associated to all clusters in the list.
-func DeleteAllMultiCloudObjects(
-	ctx context.Context, c client.Client, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, allClustersForUpgrade []string) error {
-	// Only need to delete ManagedClusterView as ManagedClusterActions get deleted automatically.
-	for _, clusterName := range allClustersForUpgrade {
-		err := DeleteMultiCloudObjects(ctx, c, clusterGroupUpgrade, clusterName)
-		if err != nil {
+// DeleteManagedClusterActions cleans up actions associated to a cluster.
```

Comment:
I would expand this and add the explanation that although MCAs get automatically deleted after 60 seconds, they can get stuck if the managed cluster becomes unavailable.